rust_src/src/symbols.rs

```diff
@@ -170,3 +171,17 @@ fn indirect_variable_lisp(object: LispObject) -> LispObject {
         object
     }
 }
+
+/// Return SYMBOL's value.  Error if that is void.  Note that if
+/// `lexical-binding' is in effect, this returns the global value
+/// outside of any lexical scope.
+#[lisp_fn]
+pub fn symbol_value(symbol: LispObject) -> LispObject {
+    unsafe {
+        let val = find_symbol_value(symbol.to_raw());
+        if val != Qunbound {
+            return LispObject::from_raw(val);
+        }
+    }
+    xsignal!(Qvoid_variable, symbol);
+}
```

Comment:
I would put this in an else block, as it only depends on the value of `val`.

Reply:
Another option would be to `== Qunbound` case first, as this will save you the explicit return :)

```rust
if val == Qunbound {
   xsignal!(Qvoid_variable, symbol);
}
LispObject::from_raw(val)
```

Reply:
I'm trying to optimize for smaller `unsafe` sections here, though that might be a bad idea. Let me know, I'm a rust newbie. Also the style point is really good and I'll refactor the code to take it into account. Before I push another commit there here are the couple ways I'm playing with doing the function:

```rust
pub fn symbol_value(symbol: LispObject) -> LispObject {
    let raw_symbol = symbol.to_raw();
    let val = unsafe { find_symbol_value(raw_symbol) };
    if val == unsafe { Qunbound } {
        xsignal!(Qvoid_variable, symbol);
    }
    LispObject::from_raw(val)
}```

That's optimizing for smallest unsafe AFAICT, I feel more comfortable leaving lots up to the compiler.

On the other hand, I could wrap the entire body in unsafe and do it naively. Is this better? Also, if I'm not mistaken, this is semantically different than marking the function itself unsafe, correct?

```rust
```

That streamlines the code a little bit by inlining the `to_raw` call, unsure of implications on compilation in that case. However, it also causes a warning to be generated since `xsignal!` does unsafe already and that block becomes redundant.

So, after typing this all out I like the idea of reversing the check and making tiny unsafe blocks. But perhaps I've misjudged something. Let me know!

Reply:
Perhaps like `constant_nil` we should have more accessible Q constants so we do not need the ugly unsafe all the time. In some of my PR I have taken to making local variables out of them.

Reply:
Wrappers seem like a good idea. For example, I am planning to port `find_symbol_value` and associated functions as I get more familiar with the codebase. If it's a priority I can add the `Qunbound` abstraction in this one, though.

Reply:
Adding on first use appears to be the plan of action. :+1:

Reply:
Looking at `constant_nil`, its contents are not marked `unsafe` while `constant_t` is. Can you clarify why that is the case?